coral-common/src/main/java/com/linkedin/coral/common/transformers/SqlCallTransformer.java

```diff
@@ -100,6 +109,18 @@ protected RelDataType getRelDataType(SqlNode sqlNode) {
         return sqlValidator.getValidatedNodeType(dummySqlSelect).getFieldList().get(0).getType();
       } catch (Throwable ignored) {
       }
+    }
+    // Additional attempt to derive data type of the input sqlNode by validating the topSelectNode, especially in cases
+    // where the input is defined as an alias in topSelectNode's selectList.
+    // (e.g. topSelectNode: Select a AS tmp FROM foo where tmp > 5).
+    // Previous attempts would try validating dummySqlNode: SELECT tmp FROM foo WHERE tmp > 5, and fail.
+    try {
+      final SqlSelect dummySqlSelect = new SqlSelect(topSelectNodes.get(0).getParserPosition(), null,
+          SqlNodeList.of(sqlNode), topSelectNodes.get(0), null, null, null, null, null, null, null);
```

Comment:
What about adding this part to the for loop above for a stronger enhancement? The `fromNode` information might be from another `topSelectNode` stored in `topSelectNodes`?

Reply:
The way im seeing the implementation, for a top-level query - "select a,b from foo" the fromNode list in that case will be - [(select a,b from foo), (foo)] 

For a complex nested query with multiple sqlSelects, I'll need to introduce a boolean to track the first (top-level) sqlSelect so that I add it to the from list twice. Also, the data type derivation logic is diff -  `sqlValidator.getValidatedNodeType(sqlNode)` . 

I prefer the current approach as the inline comment clearly explains why there's an additional attempt for type derivation and how it is different from previous attempts.